dlt/destinations/impl/filesystem/filesystem.py

```diff
@@ -41,56 +42,67 @@
     ) -> None:
-        file_name = FileStorage.get_file_name_from_file_path(local_path)
-        self.config = config
-        self.dataset_path = dataset_path
-        self.is_local_filesystem = config.protocol == "file"
+        self.client = client
+        self.table = table
+        self.is_local_filesystem = client.config.protocol == "file"
         # pick local filesystem pathlib or posix for buckets
         self.pathlib = os.path if self.is_local_filesystem else posixpath
-        self.destination_file_name = path_utils.create_path(
-            config.layout,
-            file_name,
-            schema_name,
-            load_id,
-            current_datetime=config.current_datetime,
-            load_package_timestamp=dlt.current.load_package()["state"]["created_at"],
-            extra_placeholders=config.extra_placeholders,
-        )
 
+        file_name = FileStorage.get_file_name_from_file_path(local_path)
         super().__init__(file_name)
-        fs_client, _ = fsspec_from_config(config)
-        # We would like to avoid failing for local filesystem where
-        # deeply nested directory will not exist before writing a file.
-        # It `auto_mkdir` is disabled by default in fsspec so we made some
-        # trade offs between different options and decided on this.
-        item = self.make_remote_path()
-        if self.is_local_filesystem:
-            fs_client.makedirs(self.pathlib.dirname(item), exist_ok=True)
-        fs_client.put_file(local_path, item)
+
+        if table["table_format"] == "delta":
+            import pyarrow.parquet as pq
+
+            client._write_delta_table(
+                path=self.make_remote_uri(),
+                table=pq.read_table(local_path),
+                write_disposition=table["write_disposition"],
+            )
+        else:
+            self.destination_file_name = path_utils.create_path(
+                client.config.layout,
+                file_name,
+                client.schema.name,
+                load_id,
+                current_datetime=client.config.current_datetime,
+                load_package_timestamp=dlt.current.load_package()["state"]["created_at"],
+                extra_placeholders=client.config.extra_placeholders,
+            )
+            # We would like to avoid failing for local filesystem where
+            # deeply nested directory will not exist before writing a file.
+            # It `auto_mkdir` is disabled by default in fsspec so we made some
+            # trade offs between different options and decided on this.
+            # remote_path = f"{client.config.protocol}://{posixpath.join(dataset_path, destination_file_name)}"
+            remote_path = self.make_remote_path()
+            if self.is_local_filesystem:
+                client.fs_client.makedirs(self.pathlib.dirname(remote_path), exist_ok=True)
+            client.fs_client.put_file(local_path, remote_path)
 
     def make_remote_path(self) -> str:
```

Comment:
A separation is made because Delta tables are managed at the folder level, while file formats are managed at the file level.